dracut.sh

```diff
@@ -2335,23 +2335,59 @@ else
     DRACUT_KERNEL_RD_ZSTD=
 fi
 
-if [[ $compress == $DRACUT_COMPRESS_ZSTD* && ! $DRACUT_KERNEL_RD_ZSTD ]]; then
-    dwarn "Kernel has no zstd support compiled in."
-    compress=
-fi
+
+check_kernel_compress_support() {
+    local kern_compress_config
+    case $1 in
```

Comment:
issue: The `DRACUT_COMPRESS_*` variables could be set by the user. In that case this case statement would fail.

suggestion: Tie the `kern_compress_config` with the `DRACUT_COMPRESS_*` variables (probably at the `handle compression options` section to ensure they stay in sync if compressors are added/removed).

Reply:
I'm not sure I understand how that resolves the issue. If the user sets e.g. `DRACUT_COMPRESS_BZIP2=foo` then there still is no way to know for sure which kernel config option should be checked. Probably it is `CONFIG_RD_BZIP2`, but the fact that this variable was set by the user does not necessarily mean that `compress=bzip2` as well.  And regardless of what compression type we are going to use, all of these variables will be populated so checking which ones are set does not help here.

This is why I made the `*)` case not fail fatally. If `compress=` is set to some command that we do not know, we simply warn but still try and hope for the best.

Reply:
The contract with the user is that `DRACUT_COMPRESS_BZIP2` points to a bzip2 compressor. This could be an uncommon path like `DRACUT_COMPRESS_BZIP2=foo/bar/bzip2` or some compatible command. So `CONFIG_RD_BZIP2` should be checked in that case.

Reply:
How about this?

Now it checks for the correct kernel config option for:
- `DRACUT_COMPRESS_GZIP=/usr/bin/gzip  ./dracut.sh --kver 6.10.3-gentoo-dist /tmp/test.img -f --compress='/usr/bin/gzip -9 -n' `
- `DRACUT_COMPRESS_GZIP=/usr/bin/gzip  ./dracut.sh --kver 6.10.3-gentoo-dist /tmp/test.img -f --gzip `
- `./dracut.sh --kver 6.10.3-gentoo-dist /tmp/test.img -f --compress='/usr/bin/gzip -9 -n' `
- `./dracut.sh --kver 6.10.3-gentoo-dist /tmp/test.img -f --gzip `

That should cover all cases right?

Reply:
Yes, this should cover most (the normal) cases.